datadog_checks_base/datadog_checks/base/checks/openmetrics/base_check.py

```diff
@@ -97,7 +100,27 @@ def __init__(self, *args, **kwargs):
 
         if instances is not None:
             for instance in instances:
-                self.get_scraper_config(instance)
+                possible_urls = instance.get('prometheus_possible_urls')
+                if possible_urls is not None:
+                    for url in possible_urls:
+                        try:
+                            new_instance = deepcopy(instance)
+                            new_instance.update({'prometheus_url': url})
+                            scraper_config = self.get_scraper_config(new_instance)
+                            response = self.send_request(url, scraper_config)
+                            response.raise_for_status()
+                            instance['prometheus_url'] = url
+                            self.get_scraper_config(instance)
+                            break
+                        except (IOError, requests.HTTPError, requests.exceptions.SSLError) as e:
+                            self.log.info("Couldn't connect to %s: %s, trying next possible URL.", url, str(e))
+                    else:
+                        self.log.error(
+                            "The agent could connect to none of the following URL: %s.",
+                            str(instance['prometheus_possible_urls']),
```

Comment:
```suggestion
                            str(possible_urls),
```

and perhaps `str` is not need too?

Reply:
Good point. And I agree that `str` shouldn’t be needed.
I’ve committed the change. 👍🏻